daemon/cmd/ipam.go

```diff
@@ -193,6 +194,24 @@ func (d *Daemon) allocateRouterIPv6(family types.NodeAddressingFamily) (net.IP,
 	}
 }
 
+// Coalesce CIDRS when allocating the DatapathIPs and healthIPs. GH #18868
+func coalesceCIDRs(rCIDRs []string) (newresult []string) {
+	cidrs := make([]*net.IPNet, 0, len(rCIDRs))
+	for _, k := range rCIDRs {
+		var s net.IPNet
+		ip, mask, _ := net.ParseCIDR(k)
+		s.IP = ip
+		s.Mask = mask.Mask
+		cidrs = append(cidrs, &s)
```

Comment:
nit: I think we can make this a bit more readable with:

```suggestion
		ip, mask, _ := net.ParseCIDR(k)
		cidrs = append(cidrs, &net.IPNet{IP: ip, Mask: mask.Mask})
```

Reply:
Changed as per suggestion, thank you